build-tools/automation/azure-pipelines.yaml

```diff
@@ -479,6 +479,15 @@ stages:
         artifactSource: bin/Test$(ApkTestConfiguration)/Xamarin.Android.EmbeddedDSO_Test-Signed.apk
         artifactFolder: Default
 
+    - template: yaml-templates/apk-instrumentation.yaml
+      parameters:
+        configuration: $(ApkTestConfiguration)
+        testName: Xamarin.Android.Net-Tests
```

Comment:
@simonrozsival: Why add a new app instead of adding the new tests to the existing `tests/Mono.Android-Tests` app?  I don't understand the rationale.

Reply:
I started with a test in `tests/Mono.Android-Tests` but I wasn't able to get it woring. I was getting `Java.Lang.ClassNotFoundException : xamarin.android.net.X509TrustManagerWithValidationCallback` because a java wrapper for the class I added, `X509TrustManagerWithValidationCallback`, isn't being generated automatically. I got [some tips from you](https://discord.com/channels/732297728826277939/732297837953679412/936242135286575124) and I instead created an apk project which allowed me to run the test successfully.

I reverted the apk project. I think I somehow need to special-case the `X509TrustManagerWithValidationCallback` type so that it's built into `mono.android.jar` to make the test work inside of `tests/Mono.Android-Tests`.